### DISCUSSION TITLE
How to version and publish a vscode extension?

### DISCUSSION THREAD

**Discussion by jindong-zhannng** [changesets/changesets#1297] [Q&A] [answered]
Hi team, I am building a vscode extension and wanna to leverage changesets to manage changelog and version for it.

I didn't find any guide relates to publish package to vscode registry instead of npm. I already learned this tool can work with npm well e.g. checking remote latest version from npm before publishing. But I am not sure will these features work with vscode registry as well. Any guidelines? Thanks!

**Comment by Andarist** [comment, marked as answer]
The key is to tag the extension before publishing: https://github.com/mdx-js/mdx-analyzer/blob/3069af7ade2feea2a65ad9ddcf241ceec55ec314/scripts/tag-extension.mjs

Then you can do this before calling `changeset publish`:
https://github.com/mdx-js/mdx-analyzer/blob/3069af7ade2feea2a65ad9ddcf241ceec55ec314/release.sh#L6-L7

And you need to wire it up with `changesets/action`:
https://github.com/mdx-js/mdx-analyzer/blob/3069af7ade2feea2a65ad9ddcf241ceec55ec314/.github/workflows/version.yml#L23-L24

**Reply by evoactivity** [reply]
Hi @Andarist 

Can you provide a few more details on the above.

`tag-extension.mjs` just seems to output ``console.log(`\nNew tag: ${tag}`)``. Does `npx changeset publish` use the output of the previous command to decide what the new tag should be?

`npx changeset publish` is being run in the `release.sh` file and then vsce and ovsx handle publishing to the marketplaces. Is the `changeset publish` command required for repos that _only_ publish to the extension marketplaces. Doesn't that command handle publishing to NPM as well as creating the github release? Or can the NPM part be disabled somehow?

**Reply by almahdi404** [reply]
**mdx-analyzer** is also publishing a package to npm. I guess you can skip `npx changeset publish` if you are just publishing an extension.

**Comment by theoephraim** [comment]
@Andarist - Maybe I missed something in the docs, but is there another way to tell changesets not to publish a package to npm other than marking it as private? I still want it to manage the versioning and tagging, but just want it to skip publishing to npm. The only related settings I saw was the `privatePackages` setting, but I dont want to mark my vscode extension as private, and I even if I did, I wouldn't want to affect other private packages within my monorepo.

If I understand correctly, the desired workflow would be to let changesets manage the version number and create a release tag, and then could use an additional github action to detect that release tag and publish to the extension marketplace(s). That sound right?

Thanks!
